### DISCUSSION TITLE
What is the plan?

### DISCUSSION THREAD

**Discussion by jamesaoverton** [INCATools/semantic-sql#16] [General]
TODO: Better subject line.

TLDR: We need a better way to coordinate on this stuff.

I'm happy to see Chris using RDFTab for a bunch of things. He has made a bunch of suggestions for significant changes. I'm on board, but we need to coordinate better.

My key use cases are fast and lightweight term browser and MIREOT implemented in [Gizmos](https://github.com/ontodev/gizmos). I'm happy to make changes to RDFTab even if it requires major changes to Gizmos. I'm not clear on every one else's use cases.

The SQL in this repository is really interesting, but right now I can't see using it for myself. On the other hand I use stanzas all the time, and you're not using stanzas at all, which makes me wonder how similar our needs really are.

Right now I have two people working on extending RDFTab for [Thick Triples](https://docs.google.com/spreadsheets/d/19zS8lHUM5cU_Nf9Rc7-TGL6wesOD8JLINJSan3DmPqE/edit#gid=1703761456). It's almost ready to use. The first idea is to collapse nested blank-node structures such as OWL logic into JSON, so we rarely have to deal with blank nodes. This eliminates the need for a 'stanza' column. The second idea is to put RDF reifications and OWL annotations in the same row as the triple, kind of like RDF*. Example (adjusted for https://github.com/ontodev/rdftab.rs/issues/16#issuecomment-839749257):

subject | predicate | object | datatype | annotations | metadata
-- | -- | -- | -- | -- | --
ex:foo | rdfs:subClassOf | {"rdf:type":[{"object":"owl:Restriction"}],"owl:onProperty":[{"object":"ex:part-of"}],"owl:someValuesFrom":[{"object":"ex:bar"}]} | rdf-json | {"rdfs:comment":[{"object":"OWL axiom annotation","datatype":"@en"}]} | {"rdfs:comment":[{"object":"RDF metadata","datatype":"@en"}]}

A third idea is a JSON translation of OWL Functional Syntax which I'm calling "OWL Functional S-Expressions". Example: `["ObjectSomeValuesFrom","ex:part-of","ex:bar"]`.

It's not clear to me that Chris is interested in any of these things, which is fine.

On the other hand, I'm very interested in hearing more about your ideas for reasoners. I'm also wondering if it makes sense to reimplement `relation-graph` in SQL or something, at some point.

I don't want to take on too much, but there is an opportunity here to do a bunch of things better, such as prefix management, file distribution, etc.

So what's the plan? Or at least how do we start coordinating on a plan?
